core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -130,12 +134,28 @@ private static FeedItem feedContainsFile(Feed feed, String filename) {
 
     private static FeedItem createFeedItem(Feed feed, DocumentFile file, Context context) {
         String uuid = UUID.randomUUID().toString();
-        FeedItem item = new FeedItem(0, file.getName(), uuid, file.getName(), new Date(),
-                FeedItem.UNPLAYED, feed);
-        item.setAutoDownload(false);
 
         MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
         mediaMetadataRetriever.setDataSource(context, file.getUri());
+        String dateStr = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DATE);
+
+        Date date = null;
+        if (!TextUtils.isEmpty(dateStr)) {
+            try {
+                SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd'T'HHmmss", Locale.getDefault());
+                date = simpleDateFormat.parse(dateStr);
+            } catch (ParseException parseException) {
```

Comment:
The `METADATA_KEY_DATE` format is not specified, so I think it is good to try falling back to `DateUtils.parse` here. Only if that one returns null either, I would use the file last modified date.

Reply:
fixed